eva-accession-release-automation/run_release_in_embassy/analyze_vcf_validation_results.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+# Copyright 2019 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import click
+import glob
+import logging
+import sys
+
+from ebi_eva_common_pyutils.command_utils import run_command_with_output
+from run_release_in_embassy.release_metadata import vcf_validation_output_file_pattern, asm_report_output_file_pattern
+
+logger = logging.getLogger(__name__)
+
+
+def analyze_vcf_validation_files(vcf_validation_report_files):
+    exit_code = 0
+    vcf_validation_report_error_classes_to_ignore = ["Error: Duplicated variant",
+                                                     "Warning: Reference and alternate alleles "
+                                                     "do not share the first nucleotide",
+                                                     "the input file is not valid",
+                                                     "the input file is valid",
+                                                     "not listed in a valid meta-data ALT entry"]
+    vcf_validation_error_grep_command_chain = " | ".join(['grep -v "{0}"'.format(error_class) for error_class in
+                                                          vcf_validation_report_error_classes_to_ignore])
+    for vcf_validation_report_file in vcf_validation_report_files:
+        logger.info("Analyzing file {0} ....".format(vcf_validation_report_file))
+        command_to_run = "cat {0} | {1} | wc -l".format(vcf_validation_report_file,
+                                                        vcf_validation_error_grep_command_chain)
```

Comment:
I'm not sure it's better or more efficient, but I prefer `grep -P`

```suggestion
    vcf_validation_error_grep = "|".join([vcf_validation_report_error_classes_to_ignore])
    for vcf_validation_report_file in vcf_validation_report_files:
        logger.info("Analyzing file {0} ....".format(vcf_validation_report_file))
        command_to_run = "grep -cvP '{0}' {1}".format(vcf_validation_error_grep, vcf_validation_report_file)
```

Reply:
I am not very well versed with PCRE regexps and I am wary of semicolons in the text being interpreted differently. So I would like to leave the current command chain as it is.